### PR TITLE
AIs can now cryo, going into cryogenics now has a radio message.

### PR DESCRIPTION
**Pull request by ghost**
# Github documenting your Pull Request
This **was tested.** This ports [**this PR.**](https://github.com/BeeStation/BeeStation-Hornet/pull/677)

The Automated Announcement System will announce that "[x], [x's job] has been moved into cryo storage." This prevents the frantic panic that happens when nobody hears from the DC'd captain without knowing they cryo'd. This **should also** free up the AI slot for another person to take.

# Wiki Documentation
The Automated Announcement System now announces when someone gets cryo'd.
AIs have a new "Hibernate" button in the OOC part of the statpanel.

# Changelog

:cl: MarkSuckerberg
add: "Hibernate" verb in the OOC tab for AIs, basically just the equivalent of cryo
add: the Automated announcement system now announces when someone has been cryo'd.
/:cl:

### CODE REVIEW

**Comment by JamieD1** [comment]
Change to Hibernate. Wipe core seems rather suicidal.

**Comment by ghost** [comment]
> Change to Hibernate. Wipe core seems rather suicidal.

~~Give me a moment.~~ Done.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -356,6 +356,38 @@
 /mob/living/silicon/ai/cancel_camera()
 	view_core()
 
+/mob/living/silicon/ai/verb/wipe_core()
+	set name = "Hibernate"
+	set category = "OOC"
+	set desc = "Wipe your core. This is functionally equivalent to cryo, freeing up your job slot."
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
	set desc = "Put yourself into hibernation. This is functionally equivalent to cryo, freeing up your job slot."
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -356,6 +356,38 @@
...
+
+	// Guard against misclicks, this isn't the sort of thing we want happening accidentally
+	if(alert("WARNING: This will immediately wipe your core and ghost you, removing your character from the round permanently (similar to cryo). Are you entirely sure you want to do this?",
+					"Wipe Core", "No", "No", "Yes") != "Yes")
+		return
+
+	// We warned you.
+	var/obj/structure/AIcore/latejoin_inactive/inactivecore = New(loc)
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
	var/obj/structure/AIcore/latejoin_inactive/inactivecore = new(get_turf(src))
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 3]
```diff
@@ -356,6 +356,38 @@
...
+		announcer.announce("AIWIPE", real_name, mind.assigned_role, list())
+
+	SSjob.FreeRole(mind.assigned_role)
+
+	if(mind.objectives.len)
+		mind.objectives.Cut()
+		mind.special_role = null
+
+	if(!get_ghost(1))
+		if(world.time < 30 * 600)//before the 30 minute mark
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
		if(world.time < 30 MINUTES)//before the 30 minute mark
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 4]
```diff
@@ -356,6 +356,38 @@
...
+	var/obj/structure/AIcore/latejoin_inactive/inactivecore = New(loc)
+	transfer_fingerprints_to(inactivecore)
+
+	if(GLOB.announcement_systems.len)
+		var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+		announcer.announce("AIWIPE", real_name, mind.assigned_role, list())
+
+	SSjob.FreeRole(mind.assigned_role)
+
+	if(mind.objectives.len)
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]: Mind is tied to the mob so I don't see what this does.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 5]
```diff
@@ -356,6 +356,38 @@
...
+		mind.objectives.Cut()
+		mind.special_role = null
+
+	if(!get_ghost(1))
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
	if(!get_ghost(TRUE))
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 6]
```diff
@@ -356,6 +356,38 @@
...
+		if(world.time < 30 * 600)//before the 30 minute mark
+			ghostize(0) // Players despawned too early may not re-enter the game
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
			ghostize(FALSE) // Players despawned too early may not re-enter the game
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 7]
```diff
@@ -356,6 +356,38 @@
...
+	else
+		ghostize(1)
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
		ghostize(TRUE)
```

**Review by Bibby0110 — COMMENTED** [review]
Why not offer the AI and only delete if nobody takes it?

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 8]
```diff
@@ -356,6 +356,38 @@
 /mob/living/silicon/ai/cancel_camera()
 	view_core()
 
+/mob/living/silicon/ai/verb/wipe_core()
+	set name = "Hibernate"
+	set category = "OOC"
+	set desc = "Put yourself into hibernation. This is functionally equivalent to cryo, freeing up your job slot."
+
+	// Guard against misclicks, this isn't the sort of thing we want happening accidentally
+	if(alert("WARNING: This will immediately wipe your core and ghost you, removing your character from the round permanently (similar to cryo). Are you entirely sure you want to do this?",
```
Comment (adamsong) [inline code/modules/mob/living/silicon/ai/ai.dm]: Still says wipe core

**Review by adamsong — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 8 above)
Comment (ghost) [reply code/modules/mob/living/silicon/ai/ai.dm]: I feel like it describes it well, but I'll change it.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 9]
```diff
@@ -356,6 +356,32 @@
...
 	view_core()
 
+/mob/living/silicon/ai/verb/wipe_core()
+	set name = "Hibernate"
+	set category = "OOC"
+	set desc = "Put yourself into hibernation. This is functionally equivalent to cryo, freeing up your job slot."
+
+	// Guard against misclicks, this isn't the sort of thing we want happening accidentally
+	if(alert("WARNING: This will immediately ghost you, removing your character from the round permanently (similar to cryo). Are you entirely sure you want to do this?",
+					"Wipe Core", "No", "No", "Yes") != "Yes")
```
Comment (adamsong) [inline code/modules/mob/living/silicon/ai/ai.dm]: Still says wipe core

**Review by adamsong — DISMISSED** [review]

**Comment by JamieD1** [comment]
@jc-denthead Proc names are still wipe core

**Comment by ghost** [comment]
> @jc-denthead Proc names are still wipe core

~~Give me a minute.~~

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**